src/functions.js

```diff
@@ -12,45 +18,117 @@ function getNumberOfBikes( document ) {
 };
 
 function getAllButtonText( document ) {
-  // get the text inside all tags with class 'btn' and return as an array of strings
-  
+  var buttonText = document.getElementsByClassName('btn');// get the text inside all tags with class 'btn'
+    var buttonList = [];
+
+    for (i=0; i<buttonText.length; i++){
+     
+      buttonList.push(buttonText[i].innerHTML.trim());
+
+    } // and return as an array of strings
+ return buttonList;
+ 
   // https://developer.mozilla.org/en-US/docs/Web/API/HTMLCollection
   // also you may want to convert the HTMLCollection to an array
+  
 };
 
 function getNavLinksText( document ){
+  var navBar= document.getElementById('navbarSupportedContent');
+  var links= navBar.getElementsByTagName('a');
+  var newArray = [];
+
+  for (i=0; i<links.length; i++){
+    newArray.push(links[i].textContent.trim());
+
+  }
+return newArray;
```

Comment:
similar note as the previous function, make sure the indentation is right (vs code shoud lhave the `ctrl+k, ctrl+f` key combination to help you do that automatically and try to refactor the for loop into a map